File: Dsa-day182.cpp
class Solution {
public:
    int maximizeSquareHoleArea(int n, int m, vector<int>& hBars, vector<int>& vBars) {
        int maxH = getMaxConsecutive(hBars);
        int maxV = getMaxConsecutive(vBars);
        int side = min(maxH, maxV) + 1;
        return side * side;
    }
    
private:
    int getMaxConsecutive(vector<int>& bars) {
        if (bars.empty()) return 1;
        sort(bars.begin(), bars.end());
        int maxLen = 1;
        int curr = 1;
        for (size_t i = 1; i < bars.size(); ++i) {
            if (bars[i] == bars[i - 1] + 1) {
                ++curr;
                maxLen = max(maxLen, curr);
            } else {
                curr = 1;
            }
        }
        return maxLen;
    }
};

File: Dsa-day182.py
class Solution:
    def maximizeSquareHoleArea(self, n: int, m: int, hBars: List[int], vBars: List[int]) -> int:
        def max_consecutive(bars: List[int]) -> int:
            if not bars:
                return 1
            bars.sort()
            max_len = 1
            curr = 1
            for i in range(1, len(bars)):
                if bars[i] == bars[i - 1] + 1:
                    curr += 1
                    max_len = max(max_len, curr)
                else:
                    curr = 1
            return max_len
        
        side = min(max_consecutive(hBars), max_consecutive(vBars)) + 1
        return side * side